File: frontend/src/api/services/notifications.ts
import { request } from '../http'

/**
 * 通知接口类型定义
 */
export interface Notification {
  id: number
  user_id: number
  type: 'comment' | 'reply' | 'article_mention' | 'comment_mention' | 'like' | 'bookmark' | 'system'
  title: string
  content?: string
  actor_id?: number
  actor_name?: string
  actor_avatar?: string
  target_type?: string
  target_id?: number
  target_title?: string
  target_url?: string
  is_read: boolean
  created_at: number
}

export interface NotificationListItem {
  id: number
  type: string
  title: string
  content?: string
  actor_id?: number
  actor_name?: string
  actor_avatar?: string
  target_url?: string
  is_read: boolean
  is_important?: boolean
  created_at: number
}

/**
 * 获取通知列表
 */
export const getNotifications = (params?: {
  unread_only?: boolean
  page?: number
  page_size?: number
}) => {
  return request<{ code: number; msg?: string; data: NotificationListItem[] }>({
    url: '/notifications',
    method: 'GET',
    params
  })
}

/**
 * 获取未读通知数量
 */
export const getUnreadCount = () => {
  return request<{ code: number; data: number }>({
    url: '/notifications/unread-count',
    method: 'GET'
  })
}

/**
 * 标记单条通知为已读
 */
export const markAsRead = (notificationId: number) => {
  return request<{ code: number; msg?: string }>({
    url: `/notifications/${notificationId}/read`,
    method: 'POST'
  })
}

/**
 * 标记所有通知为已读
 */
export const markAllAsRead = () => {
  return request<{ code: number; msg?: string }>({
    url: '/notifications/read-all',
    method: 'POST'
  })
}

/**
 * 删除通知
 */
export const deleteNotification = (notificationId: number) => {
  return request<{ code: number; msg?: string }>({
    url: `/notifications/${notificationId}`,
    method: 'DELETE'
  })
}

// 默认导出
export default {
  getNotifications,
  getUnreadCount,
  markAsRead,
  markAllAsRead,
  deleteNotification
}

File: backend/app/models/schemas/notifications.py
from pydantic import BaseModel, Field
from pydantic.config import ConfigDict
from typing import Optional, Literal
from datetime import datetime


class NotificationBase(BaseModel):
    user_id: int
    type: Literal['comment', 'reply', 'article_mention', 'comment_mention', 'like', 'bookmark', 'system']
    title: str
    content: Optional[str] = None
    actor_id: Optional[int] = None
    actor_name: Optional[str] = None
    actor_avatar: Optional[str] = None
    target_type: Optional[str] = None
    target_id: Optional[int] = None
    target_title: Optional[str] = None
    target_url: Optional[str] = None


class NotificationCreate(NotificationBase):
    """创建通知"""
    pass


class NotificationUpdate(BaseModel):
    """更新通知"""
    is_read: bool = False


class Notification(BaseModel):
    """通知响应"""
    model_config = ConfigDict(from_attributes=True)
    id: int
    user_id: int
    type: str
    title: str
    content: Optional[str] = None
    actor_id: Optional[int] = None
    actor_name: Optional[str] = None
    actor_avatar: Optional[str] = None
    target_type: Optional[str] = None
    target_id: Optional[int] = None
    target_title: Optional[str] = None
    target_url: Optional[str] = None
    is_read: bool
    created_at: int


class NotificationListItem(BaseModel):
    """通知列表项"""
    id: int
    type: str
    title: str
    content: Optional[str] = None
    actor_id: Optional[int] = None
    actor_name: Optional[str] = None
    actor_avatar: Optional[str] = None
    target_url: Optional[str] = None
    is_read: bool
    created_at: int


class SystemNotificationCreate(BaseModel):
    """系统通知创建"""
    title: str
    content: str
    target_users: Optional[list[int]] = None  # None 表示全员通知
